File: leetcode/power_of_two.py
# Given an integer, write a function to determine if it is a power of two.
# Time Complexity: O(n)
# Space Complexity: O(1)

class Solution:
    # @param {integer} n
    # @return {boolean}
    def isPowerOfTwo(self, n):
        return n & (n-1) == 0 and n > 0
                

if __name__ == "__main__":
    solution = Solution()
    print solution.isPowerOfTwo(1)

File: leetcode/power_of_two.js
/* Given an integer, write a function to determine if it is a power of two.
 * Time Complexity: O(1)
 * Space Complexity: O(1)
 */

/**
 * @param {number} n
 * @return {boolean}
 */
var isPowerOfTwo = function(n) {
    return ((n & (n-1)) === 0) && (n > 0) ? true : false
};
